core/src/main/java/com/backblaze/b2/util/B2Clock.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.util;
+
+/**
+ * This class provides access to two different time measurements.
+ *
+ * It also provides access to the global clock object.  It's an object
+ * instead of static methods so we can use a different implementation
+ * in tests.
+ *
+ * It is an abstract class instead of an interface so that it can hold the
+ * global instance. (If we had a dependency injection system we would
+ * probably use that instead of a global, but we don't have one.)
+ *
+ * I also keep thinking that maybe there should be two different types
+ * of clock objects -- one for monotonic time and one for wall clock time.
+ * So far, I'm keeping them together because it helps to advance them
+ * both in tests and every call to get a value says what type of time it is.
+ */
+public abstract class B2Clock {
+    // only access with while synchronized(B2Clock.class).
```

Comment:
The @GuardedBy annotation can be used to say this.

Reply:
brian and i talked and decided to not use those annotations since they require another jar.  we may decide to put the source for the annotations in our tree...maybe, if it's ok with the licensing.